rewriter/Minitest.cc

```diff
@@ -480,49 +613,173 @@ ast::ExpressionPtr runSingle(core::MutableContext ctx, bool isClass, ast::Send *
+                              std::move(moduleName));
+    }
+
+    // Handle it/xit blocks with 0 arguments separately
+    if (send->fun == core::Names::it() || send->fun == core::Names::xit()) {
+        ConstantMover constantMover;
+        ast::TreeWalk::apply(ctx, constantMover, block->body);
+
+        core::NameRef name;
+        if (send->numPosArgs() == 1) {
+            auto argString = to_s(ctx, send->getPosArg(0));
+            if (send->fun == core::Names::xit()) {
+                name = ctx.state.enterNameUTF8("<xit '" + argString + "'>");
+            } else {
+                name = ctx.state.enterNameUTF8("<it '" + argString + "'>");
             }
-            auto argLiteral = ast::cast_tree_nonnull<ast::Literal>(arg);
-            if (!argLiteral.isName()) {
-                return nullptr;
+        } else {
+            // no argument provided, use a deterministic unique name based on location
+            auto pos = send->loc.beginPos();
+            if (send->fun == core::Names::xit()) {
+                name = ctx.state.enterNameUTF8("<anonymous_xit_" + to_string(pos) + ">");
+            } else {
+                name = ctx.state.enterNameUTF8("<anonymous_it_" + to_string(pos) + ">");
             }
+        }
+
+        const bool bodyIsClass = false;
+        auto declLoc = declLocForSendWithBlock(*send);
+        auto method = addSigVoid(
+            ast::MK::SyntheticMethod0(send->loc, declLoc, std::move(name),
+                                      prepareBody(ctx, bodyIsClass, std::move(block->body), insideDescribe)));
+
+        // Only add the argument copy if there's an argument
+        if (send->numPosArgs() == 1) {
+            method = ast::MK::InsSeq1(send->loc, send->getPosArg(0).deepCopy(), move(method));
+        }
+        return constantMover.addConstantsToExpression(send->loc, move(method));
+    }
+
+    // Handle cases with 1 argument that weren't covered in the switch
+    if (send->numPosArgs() != 1) {
+        return nullptr;
+    }
+    auto &arg = send->getPosArg(0);
+
+    if (send->fun == core::Names::its()) {
+        auto argString = to_s(ctx, arg);
+        ConstantMover constantMover;
+        ast::TreeWalk::apply(ctx, constantMover, block->body);
+
+        auto name = ctx.state.enterNameUTF8("<its " + argString + ">");
+
+        const bool bodyIsClass = false;
+        auto declLoc = declLocForSendWithBlock(*send);
+        auto method = addSigVoid(
+            ast::MK::SyntheticMethod0(send->loc, declLoc, std::move(name),
+                                      prepareBody(ctx, bodyIsClass, std::move(block->body), insideDescribe)));
+        method = ast::MK::InsSeq1(send->loc, send->getPosArg(0).deepCopy(), move(method));
+        return constantMover.addConstantsToExpression(send->loc, move(method));
+    } else if (insideDescribe && ((send->fun == core::Names::let() || send->fun == core::Names::letBang())) &&
+               ast::isa_tree<ast::Literal>(arg)) {
```

Comment:
I have pulled out support for `let!` and `subject`, which also affects minitest (and **should** also affect minitest) into a separate PR, so that it can be tested and reviewed independently.

#9278